httpx/http_service.go

```diff
@@ -126,7 +149,8 @@ func (t *RetryTransport) RoundTrip(ctx context.Context, req *http.Request) (*htt
 		if resp.StatusCode >= 500 {
 			return nil, &RetryableHTTPError{Path: req.URL.String(), StatusCode: resp.StatusCode}
```

Comment:
This logic is handled by the Retry module. After the retrier decides no more calls should be made under its BackOffOpts (eg timeout), it returns both the value and the error from the most recent call. https://github.com/remind101/pkg/blob/master/retry/retry.go#L79-L82. The error will be non-nil.